project/src/main/java/frc/robot/commands/auto/AutoDrive.java

```diff
@@ -1,67 +1,99 @@
 package frc.robot.commands.auto;
 import frc.robot.Robot;
 import edu.wpi.first.wpilibj.Timer;
+//import edu.wpi.first.wpilibj.simulation.BuiltInAccelerometerSim;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.BuiltInAccelerometer;
 
 
-public class AutoDrive extends CommandBase {
-    @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
-
-    private Robot robot;
-    private Timer timer;
-    private double speed;
-    private double turn; 
-    private double duration; 
-
+public class AutoDrive extends CommandBase
+{
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+  private Robot robot;
   
   // CONSTANTS //
-  
-  
+  //private double speed;
+  //private double degrees;
+  //private double targetAngle; 
+  private boolean endCommand; 
+  private Timer driveTimer; 
+  private BuiltInAccelerometer accel; 
+  private static final double DRIVE_TIME = 0.00002;
+  private static double distanceInFeet;  
+  public static boolean onChargingStation = false;  
+
+  //private static final double TURN_SPEED = 0.5;
   
   // VARIABLES //
   
   
   
-  public AutoDrive(Robot robot, double speed, double turn, double duration)
+  public AutoDrive(Robot robot, double distanceInFeet)
   {
     this.robot = robot;
-    timer = new Timer();
-    this.speed = speed;
-    this.turn = turn;
-    this.duration = duration; 
-
+    this.distanceInFeet = distanceInFeet;
+    //this.speed = speed;
+    //this.degrees = degrees; 
+    driveTimer = new Timer();
+    driveTimer.reset(); 
+    accel = new BuiltInAccelerometer();
     addRequirements(robot.drivetrain);
+
+    
+    // Use addRequirements() here to declare subsystem dependencies.
+    
+    //addRequirements(robot.mySubsystem);
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize()
   {
-    timer.start();
-    timer.reset(); 
+   //targetAngle = robot.drivetrain.getGyroAngle()+degrees;
+   endCommand = false; 
+   SmartDashboard.putString("Robot Mode:", "AUTO DRIVE");
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute()
   {
-    while(timer.get()<duration){
-        robot.drivetrain.moveMotors(speed, turn);
+    driveTimer.start();
+    double distance = Units.feetToMeters(distanceInFeet);
+    while(driveTimer.get()>=0&&driveTimer.get()<=DRIVE_TIME){
+      if(accel.getZ()!=0){
+        onChargingStation = true;
```

Comment:
Be careful that this may false trip when your robot even tips up a little bit, it’s Z acceleration is also likely never exactly zero. I would compare it to a threshold if you can figure that out